src/sink/elasticsearch.rs

```diff
@@ -287,12 +253,36 @@ impl Sink for Elasticsearch {
                                     reason
                                 );
                             }
-                            ApiError::Other { .. } => {
+                            ApiError::DocumentMissing {
+                                ref index, ..
+                            } => {
+                                ELASTIC_ERROR_API_DOCUMENT_MISSING
+                                    .fetch_add(1, Ordering::Relaxed);
+                                error!(
+                                    "Unable to write, API Error (Document Missing): {}",
+                                    index
+                                );
+                            }
+                            ApiError::IndexAlreadyExists {
+                                ref index, ..
+                            } => {
+                                ELASTIC_ERROR_API_INDEX_ALREADY_EXISTS
+                                    .fetch_add(1, Ordering::Relaxed);
+                                error!(
+                                    "Unable to write, API Error (Index Already Exists): {}",
+                                    index
+                                );
+                            }
+                            _ => {
                                 ELASTIC_ERROR_API_UNKNOWN
                                     .fetch_add(1, Ordering::Relaxed);
                                 error!("Unable to write, API Error (Unknown)");
                             }
                         }
+                    },
+                    error::Error::Client(ref client_error) => {
+                        ELASTIC_ERROR_CLIENT.fetch_add(1, Ordering::Relaxed);
+                        error!("Unable to write, client error: {}", client_error.description());
```

Comment:
I don't see a json error handling.
Am I right saying that now we have only 2 elastic error types: Error::Api and Error::Client?

Reply:
@gliush yep, that's correct. The elastic library changed how it structures errors on us.